src/utils/package-manager.ts

```diff
@@ -0,0 +1,61 @@
+/* eslint-disable @typescript-eslint/typedef */
+import { exec } from 'child_process';
+import * as fs from 'fs';
+import * as path from 'path';
+import { promisify } from 'util';
+
+export interface PackageManager {
+  name: string;
+  label: string;
+  lockFile: string;
+  installCommand: string;
+  buildCommand: string;
+}
+
+const bun: PackageManager = {
+  name: 'bun',
+  label: 'Bun.js',
+  lockFile: 'bun.lockb',
+  installCommand: 'bun add',
+  buildCommand: 'bun build',
+};
+const yarn: PackageManager = {
+  name: 'yarn',
+  label: 'Yarn',
+  lockFile: 'yarn.lock',
+  installCommand: 'yarn add',
+  buildCommand: 'yarn build',
+};
+const pnpm: PackageManager = {
+  name: 'pnpm',
+  label: 'PNPM',
+  lockFile: 'pnpm-lock.yaml',
+  installCommand: 'pnpm add',
+  buildCommand: 'pnpm build',
+};
+const npm: PackageManager = {
+  name: 'npm',
+  label: 'NPM',
+  lockFile: 'package-lock.json',
+  installCommand: 'npm add',
+  buildCommand: 'npm run build',
+};
+
+export const packageManagers = [bun, yarn, pnpm, npm];
+
+export function detectPackageManger(): PackageManager {
+  for (const packageManager of packageManagers) {
+    if (fs.existsSync(path.join(process.cwd(), packageManager.lockFile))) {
+      return packageManager;
+    }
+  }
+  // We make the default NPM - it's weird if we don't find any lock file
+  return npm;
```

Comment:
m: I think the problem here are monorepos. For monorepo users, we expect folks to run the wizard in the respective package and not in the repo root. However, the lock file will most likely be in the root. 

I'd rather return `undefined`  and fall back to asking users about their package managers if we're actively installing packages. To print instructions it's totally fine IMO to fall back to NPM. Suggestion:
* `detectPackageManager` returns `undefined` if we don't find a lock file
* `detectPackageManagerWithFallback` (or something along these lines) returns NPM as the default. 

The alternative would be to be smarter about looking up a lock file but I think this is not trivial to solve reliably. Maybe we try this in a follow-up PR?

Reply:
Other idea: We can use similar logic to what the `detect-package-manager` package does: 

https://www.npmjs.com/package/detect-package-manager

1. check lock files
2. if none found, try to exec package manager commands
3. if none found, fall back to npm

wdyt?